src/amltk/evalutors/evaluation_protocol.py

```diff
@@ -0,0 +1,59 @@
+"""Evaluation protocols for how a trial and a pipeline should be evaluated.
+
+TODO: Sorry
+"""
+from __future__ import annotations
+
+from collections.abc import Callable, Iterable
+from typing import TYPE_CHECKING
+
+from amltk.scheduling import Plugin
+
+if TYPE_CHECKING:
+    from amltk.optimization import Trial
+    from amltk.pipeline import Node
+    from amltk.scheduling import Scheduler, Task
+
+
+class EvaluationProtocol:
+    """A protocol for how a trial and a pipeline should be evaluated."""
```

Comment:
Yup you're right. I don't have concrete implementations yet so it may change. You can imagine one implementation would be `SklearnCVEvaluator` which does a typical sklearn evaluation